scilpy/image/operations.py

```diff
@@ -112,396 +113,463 @@ def _validate_float(x):
+    if len(input_list) == 1 \
+            and not len(input_list[0].header.get_data_shape()) > 3:
         logging.error('This operation with only one operand requires 4D data.')
         raise ValueError
 
-    in_data = np.squeeze(np.rollaxis(np.array(input_list), 0,
-                                     input_list[0].ndim+1))
-
-    return np.std(in_data, axis=-1)
+    if len(input_list[0].header.get_data_shape()) > 3:
+        return np.std(input_list[0].get_fdata(dtype=np.float64), axis=-1)
+    else:
+        mean_data = mean(input_list, ref_img)
+        output_data = np.zeros(input_list[0].header.get_data_shape())
+        for img in input_list:
+            if isinstance(img, nib.Nifti1Image):
+                data = img.get_fdata(dtype=np.float64)
+                output_data += (data - mean_data) ** 2
+                img.uncache()
+            else:
+                output_data += (img - mean_data) ** 2
+        return np.sqrt(output_data / len(input_list))
 
 
-def union(input_list):
+def union(input_list, ref_img):
     """
     union: IMGs
         Operation on binary image to keep voxels, that are non-zero, in at
         least one file.
     """
-    output_data = addition(input_list)
+    output_data = addition(input_list, ref_img)
     output_data[output_data != 0] = 1
 
     return output_data
 
 
-def intersection(input_list):
+def intersection(input_list, ref_img):
     """
     intersection: IMGs
         Operation on binary image to keep the voxels, that are non-zero,
         are present in all files.
     """
-    output_data = multiplication(input_list)
+    output_data = multiplication(input_list, ref_img)
     output_data[output_data != 0] = 1
 
     return output_data
 
 
-def difference(input_list):
+def difference(input_list, ref_img):
     """
     difference: IMG_1 IMG_2
         Operation on binary image to keep voxels from the first file that are
         not in the second file (non-zeros).
     """
     _validate_length(input_list, 2)
-    _validate_arrays(*input_list)
+    _validate_imgs(*input_list, ref_img)
 
-    output_data = copy(input_list[0]).astype(np.bool)
-    output_data[input_list[1] != 0] = 0
+    output_data = np.zeros(ref_img.header.get_data_shape(), dtype=np.float64)
+    if isinstance(input_list[0], nib.Nifti1Image):
+        data_1 = input_list[0].get_fdata(dtype=np.float64)
+    else:
+        data_1 = input_list[0]
+    if isinstance(input_list[1], nib.Nifti1Image):
+        data_2 = input_list[1].get_fdata(dtype=np.float64)
+    else:
+        data_2 = input_list[1]
 
+    output_data[data_1 != 0] = 1
+    output_data[data_2 != 0] = 0
     return output_data
 
 
-def invert(input_list):
+def invert(input_list, ref_img):
     """
     invert: IMG
         Operation on binary image to interchange 0s and 1s in a binary mask.
     """
     _validate_length(input_list, 1)
-    _validate_arrays(*input_list)
+    _validate_type(input_list[0], nib.Nifti1Image)
```

Comment:
Why is the `Nifti1Image` type only sometimes enforced, but not always?

Reply:
Some operations requires the first arg to be a array/image (invert, lower_clip, etc) and other support number and array/image in any order (addition, multiplication, etc)

Reply:
I get it for simple ops, but e.g in `difference()`, you can have either `Nifti1Image` or `np.array` as input, but here in `invert` only `Nifty1Image` is allowed, so what's the difference (no pun intended)?

Reply:
Because you can do a 1 - another image, 1 will be an array for broadcasting. Difference supports using value and image 

You cannot invert a 1, invert does not support value, just image